pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionMetaDataManager.java

```diff
@@ -99,16 +98,20 @@ public void initialize() {
                 this.functionMetaDataTopicTailer.processRequest(this.functionMetaDataTopicTailer.getReader().readNext());
             }
             this.setInitializePhase(false);
-            // schedule functions if necessary
-            this.schedulerManager.schedule();
-            // start function metadata tailer
-            this.functionMetaDataTopicTailer.start();
+            
 
         } catch (Exception e) {
             log.error("Failed to initialize meta data store", e);
             throw new RuntimeException(e);
         }
     }
+    
+    public void start() {
```

Comment:
initialize and start? what cannot be done during constructor?

Reply:
because we cannot start prior to the SchedulerManager is setup since function metadata manager can invoke the scheduler.